File: API/v1/models/newsArticle.py
from datetime import datetime

from pydantic import BaseModel


class NewsArticle(BaseModel):
    id: int | None
    title: str | None
    description: str | None
    content: str | None
    link: str | None
    imagelink: str | None
    keywords: list[str] | None
    creator: list[str] | None
    symbols: list[str] | None
    pubdate: datetime | None
    sourcename: str | None
    sentiment: str | None
    aisummary: str | None

    model_config = {"from_attributes": True}

File: API/v2/lib/stockpositive/schemas/news_article.ex
defmodule Stockpositive.Schemas.NewsArticle do
  use Ecto.Schema
  import Ecto.Changeset

  schema "news" do
    field :title, :string
    field :description, :string
    field :content, :string
    field :link, :string
    field :imagelink, :string
    field :keywords, {:array, :string}
    field :creator, {:array, :string}
    field :symbols, {:array, :string}
    field :pubdate, :naive_datetime
    field :sourcename, :string
    field :sentiment, :string
    field :aisummary, :string
  end

  def changeset(article, attrs) do
    article
    |> cast(attrs, [
      :title,
      :description,
      :content,
      :link,
      :imagelink,
      :keywords,
      :creator,
      :symbols,
      :pubdate,
      :sourcename,
      :sentiment,
      :aisummary
    ])
  end
end
